cpp/src/decisiontree/batched-levelalgo/kernels.cuh

```diff
@@ -509,45 +519,35 @@ __global__ void computeSplitClassificationKernel(
 }
 
 template <typename DataT, typename LabelT, typename IdxT, int TPB>
-__global__ void computeSplitRegressionKernel(
-  DataT* pred, DataT* pred2, DataT* pred2P, IdxT* count, IdxT nbins,
-  IdxT max_depth, IdxT min_samples_split, IdxT min_samples_leaf,
-  DataT min_impurity_decrease, IdxT max_leaves,
-  Input<DataT, LabelT, IdxT> input, const Node<DataT, LabelT, IdxT>* nodes,
-  IdxT colStart, int* done_count, int* mutex, const IdxT* n_leaves,
-  volatile Split<DataT, IdxT>* splits, void* workspace, CRITERION splitType,
-  IdxT treeid, uint64_t seed) {
+__global__ void computeSplitRegressionKernelPass1(
+  DataT* pred, IdxT* count, IdxT nbins, Input<DataT, LabelT, IdxT> input,
+  const Node<DataT, LabelT, IdxT>* nodes, IdxT colStart, IdxT treeid,
+  WorkloadInfo<IdxT>* workload_info, uint64_t seed) {
   extern __shared__ char smem[];
-  IdxT nid = blockIdx.z;
+  // Read workload info for this block
+  WorkloadInfo<IdxT> workload_info_cta = workload_info[blockIdx.x];
+  IdxT nid = workload_info_cta.nodeid;
+
   auto node = nodes[nid];
   auto range_start = node.start;
   auto range_len = node.count;
 
-  // exit if current node is leaf
-  if (leafBasedOnParams<DataT, IdxT>(node.depth, max_depth, min_samples_split,
-                                     max_leaves, n_leaves, range_len)) {
-    return;
-  }
+  IdxT offset_blockid = workload_info_cta.offset_blockid;
+  IdxT num_blocks = workload_info_cta.num_blocks;
 
   // variables
   auto end = range_start + range_len;
-  auto len = nbins * 2;
+  // auto len = nbins * 2;
```

Comment:
Can we simply remove this line instead of commenting it?

Reply:
Yes. I need to redo the regression part anyway after merging with #3845. I'll remove it then.